common/templates/context_funcs.go

```diff
@@ -848,3 +848,21 @@ func (c *Context) tmplOnlineCount() (int, error) {
 
 	return online, nil
 }
+
+func (c *Context) tmplOnlineCountBots() (int, error) {
+	if c.IncreaseCheckCallCounter("online_bots", 1) {
+		return 0, ErrTooManyCalls
+	}
+
+	botCount := 0
+	c.GS.RLock()
```

Comment:
Add `c.GS.RUnlock()` just incase shit breaks

(and also remove the lock down below offcourse)

Reply:
I actually don't understand what to change here, it's straight out referenced from tmplOnlineCount .... and c.GS.RLock() before the for range  on GuildState and thereafter c.GS.Unlock().

Reply:
yeah i might have forgotten about that aswell, if it panics within the loop here it will never unlock the guild state, and the shard would eventually get stuck on trying to update it causing that shard to die, so add a defer unlock to make sure that dosen't happen and we can recover from it

Reply:
I added the defer for c.GS.RUnlock() - c.GS.RLock() stays as it is?